HTMLSerializer.js

```diff
@@ -347,6 +413,21 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Create a function that will generate unique strings which can be used as
+   * ids.
+   *
+   * @return {Function} A funtion that generates a unique string each time it is
+   *     called.
+   */
+  generateIDGenerator() {
```

Comment:
You could also use document.getElementById to check if an id exist, and if so add a suffix. Also, adding a very-likely-to-be-unique namespace prefix would make this almost unnecessary.